### PR TITLE
Mid-fi Customer Onboarding

### PR DESCRIPTION
**Pull request by tommypoa**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

# What's new in this PR

[//]: # "Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
- Mid-fi customer `OnboardingScreen`
- `react-native-snap-carousel` package
- New `Onboarding.js` constants file to house all the texts and images used for the screen
- Icon images - slightly resized to the same proportions for proper rendering

### Online sources

[//]: # "Optional - copy links to any tutorial or documentation that was useful to you when working on this PR"
- [https://github.com/archriss/react-native-snap-carousel](https://github.com/archriss/react-native-snap-carousel)
- [https://github.com/archriss/react-native-snap-carousel/blob/master/example/src/static/entries.js](https://github.com/archriss/react-native-snap-carousel/blob/master/example/src/static/entries.js)
- [https://github.com/archriss/react-native-snap-carousel/blob/master/doc/PAGINATION.md](https://github.com/archriss/react-native-snap-carousel/blob/master/doc/PAGINATION.md)
### Related PRs

[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"
- NA
## How to review

[//]: # "The order in which to review files and what to expect when testing locally"
- Alignment to Figma
- Front-end code quality and suggestions!
   - I couldn't use some of the exact pixels specified in the Figma because of a strange box that surrounds the `Pagination` component
   - Some of the margins and widths were my attempt to make the screens responsive to smaller screens, but haven't done robust testing.
## Next steps

[//]: # "What's NOT in this PR, doesn't work yet, and/or still needs to be done"
- Cleaner starting animation! There is a bug (experienced by others in the community) where the `Carousel` component does not automatically load on app launch and you need to tap on the screen before the stuff appears.
   - The fix that I found was to force a short timeout after the component mounts...not sure how/why that fixes but it does! 
   - Was thinking of adding a fade-in animation to make the entrance prettier in conjunction with this
   - To replicate the bug, just remove the timeout portion (
- Clean up front-end/css values 
- Implementing image using FontAwesome (if illustrations will take time to come)

## Tests Performed, Edge Cases

[//]: # "Hopefully we will add a testing suite/CI soon, but until then note down the steps you took to test locally"

### Screenshots/Gifs

[//]: # "Add screenshots of expected behavior - GIFs if you're feeling fancy!"
Screenshot 2020-04-16 at 9 41 01 PM
- Short timeout (explained in Next Steps) in the gif below
onboarding_gif

CC: @wangannie @12aschen 

[//]: # "This tags in both Annies as a default. Feel free to change, or add on anyone who you should be in on the conversation."

### CODE REVIEW

File: constants/Onboarding.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      'Access our resource database to help you continue eating and living healthy',
+  },
+];
+
+export const ONBOARDING_IMAGES = [
+  require('../assets/images/search_map-marker-alt_apple-alt.png'),
+  require('../assets/images/home_shopping-basket_pepper-hot.png'),
+  require('../assets/images/star_money-bill_carrot.png'),
+  require('../assets/images/heart_balance-scale_user.png'),
+];
```
Comment (kennethlien) [inline constants/Onboarding.js:29]: Should we get these from Airtable since they're only used once?

File: screens/auth/OnboardingScreen.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    super(props);
+
+    this.state = {
+      loading: true,
+      pageIndex: 0,
+    };
+  }
+
+  get pagination() {
+    return (
```
Comment (kennethlien) [inline screens/auth/OnboardingScreen.js:33]: wait this is hella cool

File: screens/auth/OnboardingScreen.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+  navigateLogIn() {
+    this.props.navigation.navigate('LogIn');
+  }
+
+  _renderItem({ item, index }) {
+    return (
+      <JustifyCenterContainer>
+        <Image
+          source={ONBOARDING_IMAGES[index]}
+          style={{ height: 55, width: 190 }}
```
Comment (kennethlien) [inline screens/auth/OnboardingScreen.js:68]: should we switch this to %

File: screens/auth/OnboardingScreen.js [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+    this.state = {
+      loading: true,
+      pageIndex: 0,
+    };
+  }
+
+  get pagination() {
+    return (
+      // TODO @tommypoa: Move styles to styled folder?
```
Comment (wangannie) [inline screens/auth/OnboardingScreen.js:34]: Yeah this would be a good idea

File: screens/auth/OnboardingScreen.js [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+      return null;
+    }
+    return (
+      <OnboardingContainer>
+        {/* Display sliding content */}
+        <Carousel
+          data={ONBOARDING_CONTENT}
+          renderItem={this._renderItem}
+          onSnapToItem={index => this.setState({ pageIndex: index })}
+          sliderWidth={300}
```
Comment (wangannie) [inline screens/auth/OnboardingScreen.js:93]: Since it has to be px, I think it would have to be `Window.width - 80`  (40px for left and right margins of `OnboardingContainer`)

Comment (wangannie) [reply screens/auth/OnboardingScreen.js:93]: To be responsive for different screen widths *

File: screens/auth/OnboardingScreen.js (hunk 3 above)
Comment (wangannie) [reply screens/auth/OnboardingScreen.js:68]: Not a huge concern rn since these are just filler images before the final designs are complete

File: constants/Onboarding.js (hunk 1 above)
Comment (tommypoa) [reply constants/Onboarding.js:29]: I feel they should be in the static folder because they are constant + pulled once, and whose space is not very large too! Want to reduce the amount of things to pull from airtable if possible in case network is an issue haha
